src/org/rocketproplab/marginalstability/flightcomputer/comm/SCMPacket.java

```diff
@@ -84,8 +88,9 @@ private void parsepacket(String packet) {
       try {
         this.id = SCMPacketType.valueOf(packetComponents[0]);
       } catch (IllegalArgumentException illegalArg) {
-        // System.out.println("Got bad packet ID (" + packetComponents[0] + ")!");
-        // TODO report error to error reporter
+        ErrorReporter errorReporter = ErrorReporter.getInstance();
+        String errorMsg = "Got bad packet ID (\" + packetComponents[0] + \")!";

```

Comment:
Once again you are printing the literal string 'Got bad packet ID (" + packetComponents[0] + ")!' but should be combining the components.